galley/pkg/config/analysis/msg/generate.main.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/ghodss/yaml"
+)
+
+// Utility for generating staticinit.gen.go. Called from gen.go
+func main() {
```

Comment:
Codegen helps with creating strongly typed things in code, so that devs can avoid silly mistakes when writing code. I think it is a well understood paradigm.